adapters/incluster/v1/apiserver.go

```diff
@@ -0,0 +1,49 @@
+package incluster
+
+import (
+	"context"
+
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+
+	clouds "github.com/kubescape/k8s-interface/cloudsupport"
+	"github.com/kubescape/k8s-interface/k8sinterface"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GetApiServerGitVersionAndCloudProvider(ctx context.Context) (string, string) {
+	k8sAPiObj := k8sinterface.NewKubernetesApi()
```

Comment:
I would eventually inject the k8sClient here in case we want to mock it - not blocking